src/main/java/com/teragrep/pth_06/planner/StreamDBClient.java

```diff
@@ -420,15 +343,17 @@ private Table<Record> getTableStatement(Condition journaldbCondition, Date day)
                     .on(JOURNALDB.LOGFILE.HOST_ID.eq(GetArchivedObjectsFilterTable.host_id).and(JOURNALDB.LOGFILE.LOGTAG.eq(GetArchivedObjectsFilterTable.tag)));
 
             if (bloomEnabled) {
-
-                selectOnConditionStep = selectOnConditionStep
-                        .leftJoin(Bloomdb.BLOOMDB.FILTER_EXPECTED_100000_FPP_001)
-                        .on(JOURNALDB.LOGFILE.ID.eq(Bloomdb.BLOOMDB.FILTER_EXPECTED_100000_FPP_001.PARTITION_ID))
-                        .leftJoin(Bloomdb.BLOOMDB.FILTER_EXPECTED_1000000_FPP_003)
-                        .on(JOURNALDB.LOGFILE.ID.eq(Bloomdb.BLOOMDB.FILTER_EXPECTED_1000000_FPP_003.PARTITION_ID))
-                        .leftJoin(Bloomdb.BLOOMDB.FILTER_EXPECTED_2500000_FPP_005)
-                        .on(JOURNALDB.LOGFILE.ID.eq(Bloomdb.BLOOMDB.FILTER_EXPECTED_2500000_FPP_005.PARTITION_ID));
-
+                Set<Table<?>> tables = walker.patternMatchTables();
```

Comment:
please elaborate what happens here

Reply:
All bloom filter tables that had a pattern match during walker traverse are left joined to the query. Dynamic number of bloom filter tables can be joined depending on number of pattern matches while keeping one to one relationship between `journaldb.logfile.id` and the bloom filter tables